video/cloud-client/analyze/beta_snippets.py

```diff
@@ -0,0 +1,203 @@
+    faces = result.annotation_results[0].face_detection_annotations
+    for i, face in enumerate(faces):
+        print('Face {}'.format(i))
+
+        # Each face_detection_annotation has only one segment.
+        segment = face.segments[0]
+        start_time = (segment.segment.start_time_offset.seconds +
+                      segment.segment.start_time_offset.nanos / 1e9)
+        end_time = (segment.segment.end_time_offset.seconds +
+                    segment.segment.end_time_offset.nanos / 1e9)
+        positions = '{}s to {}s'.format(start_time, end_time)
+        print('\tSegment: {}\n'.format(positions))
+
+        # Each detected face may appear in many frames of the video.
+        # Here we process only the first frame.
+        frame = face.frames[0]
+
+        time_offset = (frame.time_offset.seconds +
+                       frame.time_offset.nanos / 1e9)
+        box = frame.attributes[0].normalized_bounding_box
+
+        print('First frame time offset: {}s\n'.format(time_offset))
+
+        print('First frame normalized bounding box:')
+        print('\tleft  : {}'.format(box.left))
+        print('\ttop   : {}'.format(box.top))
+        print('\tright : {}'.format(box.right))
+        print('\tbottom: {}'.format(box.bottom))
+        print('\n')
+# [END video_face_bounding_boxes]
+
+
+# [START video_face_emotions]
+def face_emotions(path):
+    """ Analyze faces' emotions over frames. """
+    video_client = videointelligence.VideoIntelligenceServiceClient()
+    features = [videointelligence.enums.Feature.FACE_DETECTION]
+
+    config = videointelligence.types.FaceConfig(
+        include_emotions=True)
+    context = videointelligence.types.VideoContext(
+        face_detection_config=config)
+
+    operation = video_client.annotate_video(
+        path, features=features, video_context=context)
+    print('\nProcessing video for face annotations:')
+
+    result = operation.result(timeout=600)
+    print('\nFinished processing.')
+
+    # There is only one result because a single video was processed.
+    faces = result.annotation_results[0].face_detection_annotations
+    for i, face in enumerate(faces):
+        print('Face {}'.format(i))
+
+        frame_emotions = []
+        for frame in face.frames:
+            time_offset = (frame.time_offset.seconds +
+                           frame.time_offset.nanos / 1e9)
+            emotions = frame.attributes[0].emotions
+
+            # from videointelligence.enums
+            emotion_labels = (
+                'EMOTION_UNSPECIFIED', 'AMUSEMENT', 'ANGER',
+                'CONCENTRATION', 'CONTENTMENT', 'DESIRE',
+                'DISAPPOINTMENT', 'DISGUST', 'ELATION',
+                'EMBARRASSMENT', 'INTEREST', 'PRIDE', 'SADNESS',
+                'SURPRISE')
+
+            emotion, score = sorted(
+                [(em.emotion, em.score) for em in emotions],
+                key=lambda p: p[1])[-1]
```

Comment:
Non-trivial logic alert!

Reply:
@nnegrey I think I misunderstood your comment about this part - indeed I was sorting by emotion scores.  The reason being that the API returns one score for each emotion, and here I am trying to show only the one that scores the highest.

Reply:
added a comment here to clarify what I was doing.  please take a look.

Reply:
This line is really impressive:
 - multiple return values
 - sorting
 - list comprehension
 - lambda
 - -1 index used